azavea/temperate
# Organization Creation

## Overview

Add "create organization" wizard, with updated routing guards. If user's primary organization is not set, they will be redirected to the organization creation wizard.

### Demo

![localhost_4210_create-organization](https://user-images.githubusercontent.com/4432106/35760025-70963aae-084b-11e8-8d43-4cd6db6f715a.png)
![localhost_4210_create-organization 1](https://user-images.githubusercontent.com/4432106/35760028-771bb8cc-084b-11e8-8fbf-f78f811872a2.png)
![localhost_4210_create-organization 2](https://user-images.githubusercontent.com/4432106/35760029-7961782e-084b-11e8-9b70-658bdfce6405.png)


### Notes

 - This is based on top off PR https://github.com/azavea/temperate/pull/538 and contains commits from it, which I'll rebase away once it is merged into `develop`

 - This is a wizard is a little different form the other wizards in the project, as it doesn't save when transitioning between steps but rather when clicking "Confirm" on the last step.  I implemented it this way so that we can cleanly use the presence of `User.primary_organization` to determine whether to show the wizard or not, without worrying about the user accidentally skipping the invitation step.

 - Newly created users don't get assigned to the default organization, though the fixture for it still exists.

## Testing Instructions

 * If your user has a primary organization, unset it in the Django Admin
 * Attempt to go to any page
 * Should be redirected to `/create-organization`
 * Should be able to step through wizard
 * Should be required to set city and name
 * Should get error messages for city and name fields if they are missing
 * If an organization name longer than 256 characters is entered, clicking "Confirm" from step 3 should show an error toast, and direct the user back to step 1 with an error displayed below the field
 * If all fields are valid, clicking "Confirm" from step 3 should save the organization and go the Create a Plan wizard


Closes #483
